clients/python/test/test_osparc/test_apis.py

```diff
@@ -0,0 +1,48 @@
+import os
+
+import pytest
+from osparc import SolversApi, StudiesApi
+from osparc._models import ParentProjectInfo
+
+
+@pytest.mark.parametrize("parent_headers", [True, False])
+def test_create_jobs_parent_headers(
+    mocker, faker, enable_dev_mode, parent_headers: bool
+):
+    def check_headers(**kwargs):
+        parent_info = ParentProjectInfo()
+        if parent_headers:
+            assert parent_info.x_simcore_parent_project_uuid is not None
+            assert parent_info.x_simcore_parent_node_id is not None
+            assert parent_info.x_simcore_parent_project_uuid == kwargs.get(
+                "x_simcore_parent_project_uuid"
+            )
+            assert parent_info.x_simcore_parent_node_id == kwargs.get(
+                "x_simcore_parent_node_id"
+            )
+
+    mocker.patch(
+        "osparc_client.SolversApi.create_job",
+        side_effect=lambda solver_key, version, job_inpus, **kwargs: check_headers(
+            **kwargs
+        ),
+    )
+    mocker.patch(
+        "osparc_client.StudiesApi.create_study_job",
+        side_effect=lambda study_id, job_inputs, **kwargs: check_headers(**kwargs),
+    )
+    mocker.patch(
+        "osparc_client.StudiesApi.clone_study",
+        side_effect=lambda study_id, **kwargs: check_headers(**kwargs),
+    )
+
+    if parent_headers:
+        os.environ["OSPARC_STUDY_ID"] = f"{faker.uuid4()}"
```

Comment:
you could even check that you get the expected OSPARC_STUDY_ID... what if you somehow read from the wrong ENV into the other?
OSPARC_STUDY_ID into parent_node_id for example. this test would now not find it out right?

Reply:
Hmm, I actually think that should already be tested. But I have cleaned up the test a bit to ensure this is tested properly